From: Aron Xu <aron@debian.org>
Date: Sat, 4 Feb 2012 12:23:59 +0800
Subject: [PATCH] increase dbus max retry times

---
 src/module/dbus/dbusstuff.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/module/dbus/dbusstuff.c b/src/module/dbus/dbusstuff.c
index 165c77e..e5c3297 100644
--- a/src/module/dbus/dbusstuff.c
+++ b/src/module/dbus/dbusstuff.c
@@ -45,7 +45,7 @@ typedef struct _FcitxDBus {
 } FcitxDBus;
 
 #define RETRY_INTERVAL 2
-#define MAX_RETRY_TIMES 5
+#define MAX_RETRY_TIMES 15
 
 static void* DBusCreate(FcitxInstance* instance);
 static void DBusSetFD(void* arg);
-- 
