From: IME Packaging Team <pkg-ime-devel@lists.alioth.debian.org>
Date: Fri, 3 Feb 2012 02:40:46 +0800
Subject: [PATCH] 01-fix_wm_class_setting

===================================================================
---
 src/module/x11/x11stuff.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/src/module/x11/x11stuff.c b/src/module/x11/x11stuff.c
index 43ea391..b8f7ce9 100644
--- a/src/module/x11/x11stuff.c
+++ b/src/module/x11/x11stuff.c
@@ -293,7 +293,13 @@ void* X11SetWindowProperty(void* arg, FcitxModuleFunctionArg args)
     pid_t pid = getpid();
     XChangeProperty(x11priv->dpy, window, x11priv->pidAtom, XA_CARDINAL, 32,
                     PropModeReplace, (unsigned char *)&pid, 1);
-    XChangeProperty(x11priv->dpy, window, XA_WM_CLASS, XA_STRING, 8, PropModeReplace, (const unsigned char*) "Fcitx", strlen("Fcitx") + 1);
+    
+	char res_name[] = "fcitx";
+	char res_class[] = "fcitx";
+	XClassHint ch;
+	ch.res_name = res_name;
+	ch.res_class = res_class;
+	XSetClassHint(x11priv->dpy, window, &ch);
 
     if (windowTitle) {
         XTextProperty   tp;
-- 
