From: IME Packaging Team <pkg-ime-devel@lists.alioth.debian.org>
Date: Fri, 3 Feb 2012 02:40:47 +0800
Subject: [PATCH] 01-fix_duplicate_entry

---
 src/lib/fcitx/ime.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/lib/fcitx/ime.c b/src/lib/fcitx/ime.c
index ff27f2a..49f3885 100644
--- a/src/lib/fcitx/ime.c
+++ b/src/lib/fcitx/ime.c
@@ -248,6 +248,7 @@ void FcitxRegisterIMv2(FcitxInstance *instance,
     if (GetIMFromIMList(imes, uniqueName))
     {
         FcitxLog(ERROR, "%s already exists", uniqueName);
+        return ;
     }
     
     memset(&newime, 0, sizeof(FcitxIM));
-- 
