From: Weng Xuetian <wengxt@gmail.com>
Date: Mon, 26 Dec 2011 14:17:19 +0800
Subject: [PATCH] test DISPLAY

---
 data/script/fcitx-configtool |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/data/script/fcitx-configtool b/data/script/fcitx-configtool
index d09d70d..d34496a 100755
--- a/data/script/fcitx-configtool
+++ b/data/script/fcitx-configtool
@@ -46,12 +46,19 @@ run_xdg()
 {
     command=`which xdg-open`
     if [ $? -eq 0 ]; then
-        exec $command ~/.config/fcitx/config
+        exec $command $HOME/.config/fcitx/config
     else
         exit 0
     fi
 }
 
+if [ ! -n "$DISPLAY" ]; then
+    [ -n "$EDITOR" ] && which $EDITOR >& /dev/null && exec $EDITOR $HOME/.config/fcitx/config
+    [ -n "$VISUAL" ] && which $VISUAL >& /dev/null && exec $VISUAL $HOME/.config/fcitx/config
+    echo 'Please run it under X, or set $EDITOR or $VISUAL'
+    exit 0
+fi
+
 detectDE
 
 case "$DE" in
@@ -61,4 +68,4 @@ case "$DE" in
     *)
         run_gtk
         ;;
-esac
\ No newline at end of file
+esac
-- 
