--- a/src/lib/fcitx/ime.c
+++ b/src/lib/fcitx/ime.c
@@ -414,11 +414,13 @@
         }
     }
 
+#if 0
     /* Added by hubert_star AT forum.ubuntu.com.cn */
     if (event == FCITX_RELEASE_KEY
             && IsHotKeySimple(sym, state)
             && retVal == IRV_TO_PROCESS)
         return IRV_DO_NOTHING;
+#endif
 
     if (retVal == IRV_TO_PROCESS) {
         /* process key event for switch key */
