Index: fcitx/src/lib/fcitx/instance.c
===================================================================
--- fcitx.orig/src/lib/fcitx/instance.c	2011-12-21 19:35:48.296142191 +0800
+++ fcitx/src/lib/fcitx/instance.c	2011-12-21 19:56:06.358182227 +0800
@@ -156,25 +156,25 @@
         snprintf(strTemp, PATH_MAX, "@im=%s", imname);
         strTemp[PATH_MAX - 1] = '\0';
 
-        if ((getenv("XMODIFIERS") != NULL && CHECK_ENV("XMODIFIERS", strTemp, true)) ||
-                (CHECK_ENV("GTK_IM_MODULE", "xim", false) && CHECK_ENV("GTK_IM_MODULE", "fcitx", false))
-                || (CHECK_ENV("QT_IM_MODULE", "xim", false) && CHECK_ENV("QT_IM_MODULE", "fcitx", false))) {
-            char *msg[12];
-            msg[0] = _("Please check your environment varibles.");
-            msg[1] = _("You can use tools provided by your distribution.");
-            msg[2] = _("Or You may need to set environment varibles below to make fcitx work correctly.");
-            msg[3] = "export XMODIFIERS=\"@im=fcitx\"";
-            msg[4] = "export QT_IM_MODULE=xim";
-            msg[5] = "export GTK_IM_MODULE=xim";
-            msg[6] = _("Or (Depends on you install im module or not)");
-            msg[7] = "export XMODIFIERS=\"@im=fcitx\"";
-            msg[8] = "export QT_IM_MODULE=fcitx";
-            msg[9] = "export GTK_IM_MODULE=fcitx";
-            msg[10] = _("If you use login manager like gdm or kdm, put those lines in your ~/.xprofile.");
-            msg[11] = _("If you use ~/.xinitrc and startx, put those lines in ~/.xinitrc.");
+/*        if ((getenv("XMODIFIERS") != NULL && CHECK_ENV("XMODIFIERS", strTemp, true)) ||*/
+/*                (CHECK_ENV("GTK_IM_MODULE", "xim", false) && CHECK_ENV("GTK_IM_MODULE", "fcitx", false))*/
+/*                || (CHECK_ENV("QT_IM_MODULE", "xim", false) && CHECK_ENV("QT_IM_MODULE", "fcitx", false))) {*/
+/*            char *msg[12];*/
+/*            msg[0] = _("Please check your environment varibles.");*/
+/*            msg[1] = _("You can use tools provided by your distribution.");*/
+/*            msg[2] = _("Or You may need to set environment varibles below to make fcitx work correctly.");*/
+/*            msg[3] = "export XMODIFIERS=\"@im=fcitx\"";*/
+/*            msg[4] = "export QT_IM_MODULE=xim";*/
+/*            msg[5] = "export GTK_IM_MODULE=xim";*/
+/*            msg[6] = _("Or (Depends on you install im module or not)");*/
+/*            msg[7] = "export XMODIFIERS=\"@im=fcitx\"";*/
+/*            msg[8] = "export QT_IM_MODULE=fcitx";*/
+/*            msg[9] = "export GTK_IM_MODULE=fcitx";*/
+/*            msg[10] = _("If you use login manager like gdm or kdm, put those lines in your ~/.xprofile.");*/
+/*            msg[11] = _("If you use ~/.xinitrc and startx, put those lines in ~/.xinitrc.");*/
 
-            DisplayMessage(instance, _("Setting Hint"), msg, 12);
-        }
+/*            DisplayMessage(instance, _("Setting Hint"), msg, 12);*/
+/*        }*/
     }
     /* make in order to use block X, query is not good here */
     pthread_create(&instance->pid, NULL, RunInstance, instance);
