diff -r 2793d9cf16bb src/frontend/gtk2/CMakeLists.txt
--- a/src/frontend/gtk2/CMakeLists.txt	Wed Sep 21 11:41:25 2011 +0800
+++ b/src/frontend/gtk2/CMakeLists.txt	Wed Sep 21 11:52:32 2011 +0800
@@ -3,8 +3,9 @@
 PKG_CHECK_MODULES(GTK2 REQUIRED "gtk+-2.0" )
 
 _pkgconfig_invoke("gtk+-2.0" GTK2 BINARY_VERSION "" "--variable=gtk_binary_version")
+_pkgconfig_invoke("gtk+-2.0" GTK2 LIBDIR "" "--variable=libdir")
 set(GTK2_IM_MODULEDIR
- "${libdir}/gtk-2.0/${GTK2_BINARY_VERSION}/immodules"
+ "${GTK2_LIBDIR}/gtk-2.0/${GTK2_BINARY_VERSION}/immodules"
 )
 
 include_directories(${GTK2_INCLUDE_DIRS})
diff -r 2793d9cf16bb src/frontend/gtk3/CMakeLists.txt
--- a/src/frontend/gtk3/CMakeLists.txt	Wed Sep 21 11:41:25 2011 +0800
+++ b/src/frontend/gtk3/CMakeLists.txt	Wed Sep 21 11:52:32 2011 +0800
@@ -3,9 +3,9 @@
 PKG_CHECK_MODULES(GTK3 REQUIRED "gtk+-3.0" )
 
 _pkgconfig_invoke("gtk+-3.0" GTK3 BINARY_VERSION "" "--variable=gtk_binary_version")
-
+_pkgconfig_invoke("gtk+-3.0" GTK3 LIBDIR "" "--variable=libdir")
 set(GTK3_IM_MODULEDIR
- "${libdir}/gtk-3.0/${GTK3_BINARY_VERSION}/immodules"
+ "${GTK3_LIBDIR}/gtk-3.0/${GTK3_BINARY_VERSION}/immodules"
 )
 
 include_directories(${GTK3_INCLUDE_DIRS})
