diff -urN ../tmp-orig/fcitx-2.0.2/src/InputWindow.c ./src/InputWindow.c
--- ../tmp-orig/fcitx-2.0.2/src/InputWindow.c        2004-02-12 12:14:12.000000000 +0100
+++ ./src/InputWindow.c        2004-06-09 23:48:08.011961969 +0200
@@ -110,7 +110,7 @@
     char            str1[] = "Ay中";
     char            str2[10];
     char           *ps1, *ps2;
-    int             l1, l2;
+    size_t          l1, l2;
 
     if (!xftFont)
         return;
diff -urN ../tmp-orig/fcitx-2.0.2/src/ui.c ./src/ui.c
--- ../tmp-orig/fcitx-2.0.2/src/ui.c        2004-04-28 08:45:15.000000000 +0200
+++ ./src/ui.c        2004-06-09 23:49:20.325901311 +0200
@@ -287,7 +287,7 @@
     XGlyphInfo      extents;
     char            str1[100];
     char           *ps, *ps1;
-    int             l1, l2;
+    size_t          l1, l2;
 
     if (!xftFont)
         return 0;
@@ -326,7 +326,7 @@
 void OutputString (Window window, char *str, int x, int y, XColor color)
 {
     char            strOutput[100] = "";        //该长度应该够用了
-    int             l1, l2;
+    size_t          l1, l2;
     char           *ps;
     XftColor        xftColor;
     XRenderColor    renderColor;
diff -urN ../tmp-orig/fcitx-2.0.2/src/xim.c ./src/xim.c
--- ../tmp-orig/fcitx-2.0.2/src/xim.c        2004-04-27 18:41:34.000000000 +0200
+++ ./src/xim.c        2004-06-09 23:49:49.717186704 +0200
@@ -342,7 +342,7 @@
     char           *ps;
 
     if (bIsUtf8) {
-        int             l1, l2;
+        size_t          l1, l2;
 
         ps = strOutput;
         l1 = strlen (strHZ);
